packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -315,11 +315,28 @@ export class ProfilesCache {
         return baseProfile;
     }
 
-    // This will retrieve the base profile from imperative
-    public async fetchBaseProfile(): Promise<imperative.IProfileLoaded | undefined> {
+    /**
+     * Retrieves the base profile from Imperative to use for log in/out. If a
+     * nested profile name is specified (e.g. "lpar.zosmf"), then its parent
+     * profile is returned unless token is already stored in the base profile.
+     * @param profileName Name of profile that was selected in the tree
+     * @returns IProfileLoaded object or undefined if no profile was found
+     */
+    public async fetchBaseProfile(profileName?: string): Promise<imperative.IProfileLoaded | undefined> {
         const mProfileInfo = await this.getProfileInfo();
         const baseProfileAttrs = mProfileInfo.getDefaultProfile("base");
-        if (baseProfileAttrs == null) {
+        const config = mProfileInfo.getTeamConfig();
+        if (
+            profileName?.includes(".") &&
+            (baseProfileAttrs == null || !config.api.secure.securePropsForProfile(baseProfileAttrs.profName).includes("tokenValue"))
```

Comment:
Question about this line here.
In order for this function to return the default base profile (old behavior), the token only needs to be in the secure array of the base profile and not actually stored in the vault. Is that assumption correct?

Follow up question, Does that means that this function will return the default base profile even if I already have a token stored in the parent profile?

Reply:
Good questions, sorry I didn't see this until after the PR was merged. Your assumption is correct 😁 

For the 2nd question, that's right. I think the reason was for backwards compatibility - however we probably want to give precedence to whichever profile has tokenValue already defined 😋